onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1380,6 +1396,11 @@ TensorrtExecutionProvider::TensorrtExecutionProvider(const TensorrtExecutionProv
         ep_context_embed_mode_ = std::stoi(ep_context_embed_mode_env);
       }
 
+      // incase the EP context is dumped the engine cache has to be enabled
```

Comment:
nit: Make it a tiny lambda since we have the same conditions above in lines 1262-1265?

Reply:
modified.